sdk/core/Azure.Core/api/Azure.Core.net461.cs

```diff
@@ -984,6 +993,20 @@ public partial class HttpPipelineTransportOptions
         public System.Collections.Generic.IList<System.Security.Cryptography.X509Certificates.X509Certificate2> ClientCertificates { get { throw null; } }
         public System.Func<Azure.Core.Pipeline.ServerCertificateCustomValidationArgs, bool>? ServerCertificateCustomValidationCallback { get { throw null; } set { } }
     }
+    public abstract partial class RetryPolicy : Azure.Core.Pipeline.HttpPipelinePolicy
+    {
+        protected RetryPolicy(Azure.Core.RetryOptions? options = null) { }
+        protected virtual System.TimeSpan CalculateNextDelay(Azure.Core.HttpMessage message) { throw null; }
```

Comment:
Would this override any server-specified retry-after values?  Might be nice to word the name to clarify.

Reply:
Yes, do you have a suggestion of how to incorporate that? I think in general, retry-after, is a fairly niche thing.

Reply:
I'm still a little nervous about letting libraries override a service-specified retry-after header.  It would let people DOS services with our libraries, possibly accidentally.

Reply:
It could be implemented by calling the virtual method from inside a non-virtual method that uses the service's retry-after value if it's provided.

Reply:
My opinion is that this is an advanced feature to allow users to have more control. As we've discussed previously, users can get into bad situations when modifying/adding policies if they are not careful. The user would have access to the retry-after header so they can use it in their implementation.

Reply:
I'm not sure if we would want it to be virtual as this might make users think they need to override it. What if we go with protected static?